backend/engine/plugins/cicd_tools/main.py

```diff
@@ -0,0 +1,153 @@
+    name: str
+    pattern: str
+    validator: Optional[ValidatorFn]
+
+
+class CICDToolsDetails(TypedDict):
+    display_name: str
+    configs: list[str]
+
+
+log = utils.setup_logging("cicd_tools")
+
+
+# For CICD tools with simple config patterns, we can search with a glob pattern instead of writing a
+# full detector
+pattern_detector_configs: list[PatternDetectorConfig] = [
+    {
+        "id": "azure_pipelines",
+        "name": "Azure Pipelines",
+        "pattern": "**/.azure-pipelines.yml",
+        "validator": None,
+    },
+    {
+        "id": "circleci",
+        "name": "CircleCI",
+        "pattern": ".circleci/config.yml",
+        "validator": None,
+    },
+    {
+        "id": "github_actions",
+        "name": "GitHub Actions",
+        "pattern": ".github/workflows/*.yml",
+        "validator": None,
+    },
+    {
+        "id": "gitlab_ci",
+        "name": "GitLab CI",
+        "pattern": ".gitlab-ci.yaml",
+        "validator": None,
+    },
+    {
+        "id": "jenkins",
+        "name": "Jenkins",
+        "pattern": "**/*Jenkinsfile*",
+        "validator": None,
+    },
+    {
+        "id": "teamcity",
+        "name": "TeamCity",
+        "pattern": ".teamcity/",
+        "validator": None,
+    },
+    {
+        "id": "travis_ci",
+        "name": "Travis CI",
+        "pattern": ".travis.yml",
+        "validator": None,
+    },
+    {
+        "id": "bitrise",
+        "name": "BitRise",
+        "pattern": "**/*bitrise*.yml",
+        "validator": None,
+    },
+]
+
+
+detectors: list[Detector] = [
+    # Some detectors have more complicated detection logic. These need dedicated detectors
+    AWSCodeBuildDetector(),
+    ElectronForgeDetector(),
+]
+detectors.extend(
+    [
+        PatternDetector(config["id"], config["name"], config["pattern"], config["validator"])
+        for config in pattern_detector_configs
+    ]
+)
+
+
+def main(in_args=None):
+    args = utils.parse_args(in_args)
+
+    results = find_tools(args.path)
+    in_use_results = [item for item in results if item["in_use"]]
+
+    details = get_details(in_use_results)
+    event_info = [item["id"] for item in in_use_results]
+
+    # Use raw `results` instead of `in_use_results` for messages, since there may still be errors
+    # when `in_use=False`
+    debug, alerts, errors = get_messages(results)
+
+    print(
+        json.dumps(
+            {
+                "success": True,  # Always succeed, this is informational for now
+                "details": details,
+                "truncated": False,
+                "debug": debug,
+                "alerts": alerts,
+                "errors": errors,
+                "event_info": event_info,
+            }
+        )
+    )
+
+
+def find_tools(path: str) -> list[DetectorResult]:
+    return [detector.check(path) for detector in detectors]
+
+
+def get_details(results: list[DetectorResult]) -> dict[str, dict[str, CICDToolsDetails]]:
+    return {
+        "cicd_tools": {
```

Comment:
I know this is the JSON output I proposed for this plugin, but this was just a swag - please review just to ensure that it complies with output from similar APIs in the system... for example, `scan_options.plugins` in a scan object is just just an array of the internal plugin ids (no display name), and we maintain an internal mapping in backend & UI to the display names. Would this be a better and more consistent approach?

Reply:
I don't think a `scan_options.plugins` type mapping would make sense in this context, but I also don't have a good grasp of what you are envisioning there. At a minimum, it would have the added complexity of causing us to have to edit things in two places when we want to add additional CICD tools or modify the current ones.

The current pattern is similar to the one in the repo healthcheck plugins, where there's an ID for the particular finding and then the plugins also provide a plaintext `display_name` analogue that eventually gets displayed in the UI. It keeps everything localized in the plugin, so changing things is more intuitive, IMO.